contracts/libraries/VaultLifecycle.sol

```diff
@@ -144,32 +144,80 @@ library VaultLifecycle {
     /**
      * @notice Calculate the shares to mint, new price per share, and
       amount of funds to re-allocate as collateral for the new round
-     * @param currentShareSupply is the total supply of shares
+     * @param vaultState is the storage variable vaultState passed from RibbonVault
+     * @param currentShareSupply is the supply of the shares invoked with totalSupply()
      * @param asset is the address of the vault's asset
      * @param decimals is the decimals of the asset
-     * @param pendingAmount is the amount of funds pending from recent deposits
+     * @param lastQueuedWithdrawAmount is the amount queued for withdrawals from last round
+     * @param performanceFee is the perf fee percent to charge on premiums
+     * @param managementFee is the management fee percent to charge on the AUM
      * @return newLockedAmount is the amount of funds to allocate for the new round
      * @return queuedWithdrawAmount is the amount of funds set aside for withdrawal
      * @return newPricePerShare is the price per share of the new round
      * @return mintShares is the amount of shares to mint from deposits
+     * @return performanceFeeInAsset is the performance fee charged by vault
+     * @return totalVaultFee is the total amount of fee charged by vault
      */
     function rollover(
+        Vault.VaultState storage vaultState,
         uint256 currentShareSupply,
         address asset,
         uint256 decimals,
-        uint256 pendingAmount,
-        uint256 queuedWithdrawShares
+        uint256 lastQueuedWithdrawAmount,
+        uint256 performanceFee,
+        uint256 managementFee
     )
         external
         view
         returns (
             uint256 newLockedAmount,
             uint256 queuedWithdrawAmount,
             uint256 newPricePerShare,
-            uint256 mintShares
+            uint256 mintShares,
+            uint256 performanceFeeInAsset,
+            uint256 totalVaultFee
         )
     {
         uint256 currentBalance = IERC20(asset).balanceOf(address(this));
+        uint256 pendingAmount = vaultState.totalPending;
+        uint256 lockedAmount = vaultState.lockedAmount;
+        uint256 queuedWithdrawShares = vaultState.queuedWithdrawShares;
+        uint256 lastQueuedAmount = lastQueuedWithdrawAmount;
+
+        uint256 pricePerShareBeforeFee =
+            ShareMath.pricePerShare(
+                currentShareSupply,
+                currentBalance,
+                pendingAmount,
+                decimals
+            );
+
+        uint256 queuedWithdrawBeforeFee =
```

Comment:
should we have a ternary like we do here https://github.com/ribbon-finance/ribbon-v2/blob/853d281a7bc0fb67e18dd33f3eef5b0c33d479be/contracts/libraries/VaultLifecycle.sol#L237